### PR TITLE
LPD-26354 Prevent SPA from running when page is in excludePath

### PR DESCRIPTION
**Pull request by fortunatomaldonado**
https://liferay.atlassian.net/browse/LPP-54145
https://liferay.atlassian.net/browse/LPD-26354

SPA is still enabled when a page is added to the excludePath. SPA needs to check if the current page is in the excludePath and return before setting up SPA.

I had to use `// @ts-ignore` or else an format error would be thrown for Liferay.SPA. Let me know if this is the correct way to go about doing this.
Please let me know if there are any questions or comments about this.
Thank you!

### CODE REVIEW

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

File: modules/test/playwright/tests/frontend-js-spa-web/customExcludePathSPA.spec.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	await test.step('Go to page and check SPA', async () => {
+		await page.goto(layout.friendlyURL);
+
+		await page.reload();
+
+		// @ts-ignore
+
+		const liferaySPAOutput = await page.evaluate(() => Liferay.SPA);
+
+		expect(liferaySPAOutput).toEqual(undefined);
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-js-spa-web/customExcludePathSPA.spec.ts:57]: To improve the test and validate this logic, could we add a step on start, that makes the same check and asserts something like 
```js
expect(output).not.toEqual(undefined);
```

**Review by markocikos — COMMENTED** [review]
@fortunatomaldonado few questions and comments inline

File: modules/apps/frontend-js/frontend-js-spa-web/src/main/java/com/liferay/frontend/js/spa/web/internal/servlet/taglib/SPATopHeadJSPDynamicInclude.java [hunk 2]
```diff
@@ -49,6 +51,22 @@ public void include(
 		throws IOException {
 
 		SPAHelper spaHelper = _spaHelperSnapshot.get();
+
+		JSONArray excludedPathsJSONArray =
+			spaHelper.getExcludedPathsJSONArray();
+
+		String urlPath = _portal.getOriginalServletRequest(
+			httpServletRequest
+		).getServletPath();
```
[the remark this answers is not in the document]
Comment (fortunatomaldonado) [reply modules/apps/frontend-js/frontend-js-spa-web/src/main/java/com/liferay/frontend/js/spa/web/internal/servlet/taglib/SPATopHeadJSPDynamicInclude.java]: Yes it is, changed. Thank you!

File: modules/test/playwright/tests/frontend-js-spa-web/customExcludePathSPA.spec.ts
[the remark this answers is not in the document]
Comment (fortunatomaldonado) [reply modules/test/playwright/tests/frontend-js-spa-web/customExcludePathSPA.spec.ts:1]: Makes sense, changed!

File: modules/test/playwright/tests/frontend-js-spa-web/customExcludePathSPA.spec.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	page,
+	systemSettingsPage,
+}) => {
+	await test.step('Exclude path in SPA Settings', async () => {
+		await systemSettingsPage.goToSystemSetting(
+			'Infrastructure',
+			'Frontend SPA Infrastructure'
+		);
+
+		const customExcludePath = page.getByLabel('Custom Excluded Paths', {
```
[the remark this answers is not in the document]
Comment (fortunatomaldonado) [reply modules/test/playwright/tests/frontend-js-spa-web/customExcludePathSPA.spec.ts]: Got it, thank you!

File: modules/test/playwright/tests/frontend-js-spa-web/customExcludePathSPA.spec.ts (hunk 1 above)
Comment (fortunatomaldonado) [reply modules/test/playwright/tests/frontend-js-spa-web/customExcludePathSPA.spec.ts]: Added the check, thank you!

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Review by markocikos — APPROVED** [review]
@fortunatomaldonado LGTM! :sparkles:

**Pull request closed without merging**